packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1046,11 +1755,39 @@ class MessageProperty extends DiagnosticsProperty<void> {
   ///
   /// The [name], `message`, and [level] arguments must not be null.
   MessageProperty(String name, String message, {
+    DiagnosticsTreeStyle style = DiagnosticsTreeStyle.singleLine,
     DiagnosticLevel level = DiagnosticLevel.info,
   }) : assert(name != null),
        assert(message != null),
+       assert(style != null),
        assert(level != null),
-       super(name, null, description: message, level: level);
+       super(name, null, description: message, style: style, level: level);
+}
+
+class UrlProperty extends DiagnosticsProperty<String> {
+  /// Create a diagnostics property for describing urls.
+  ///
+  /// The [url], and [level] arguments must not be null.
+  UrlProperty(String name, {
+    @required String url,
+    DiagnosticsTreeStyle style = DiagnosticsTreeStyle.indentedSingleLine,
+    DiagnosticLevel level = DiagnosticLevel.info,
+  }) : assert(level != null),
```

Comment:
Add:

```assert(style != null);```

?

Reply:
done. fyi stay away from diagnostics.dart for now.
With more time I will drastically remove the diff. this is just the steel thread version that renders all the new styles well enough but has some really ugly code.